### PR TITLE
Support cuda graph for DP attention

### PR DESCRIPTION
**Pull request by ispobock**
## Motivation

Support cuda graph for DP attention (https://github.com/sgl-project/sglang/pull/1970).

### CODE REVIEW

**Review by zhyncs — APPROVED** [review]

File: python/sglang/srt/managers/scheduler.py [hunk 1]
```diff
@@ -411,16 +411,12 @@ def prepare_dp_attn_batch(self, local_batch: ScheduleBatch):
         else:
             num_tokens = local_batch.extend_num_tokens
 
-        local_num_tokens = torch.tensor(
-            num_tokens, dtype=torch.int64, device=self.device
-        )
-        global_num_tokens = torch.empty(
-            self.tp_size, dtype=torch.int64, device=self.device
-        )
+        local_num_tokens = torch.tensor([num_tokens], dtype=torch.int64)
```
Comment (merrymercy) [inline python/sglang/srt/managers/scheduler.py:414]:
```suggestion
        local_num_tokens = torch.tensor(num_tokens, dtype=torch.int64)
```

File: python/sglang/srt/managers/scheduler.py [hunk 2]
```diff
@@ -411,16 +411,12 @@ def prepare_dp_attn_batch(self, local_batch: ScheduleBatch):
...
+        global_num_tokens = torch.empty(self.tp_size, dtype=torch.int64)
         torch.distributed.all_gather_into_tensor(
             global_num_tokens,
             local_num_tokens,
-            group=self.tp_worker.get_tp_device_group(),
+            group=self.tp_worker.get_tp_cpu_group(),
```
Comment (merrymercy) [inline python/sglang/srt/managers/scheduler.py]:
```suggestion
            group=self.tp_cpu_group,
```

File: python/sglang/srt/managers/scheduler.py [hunk 3]
```diff
@@ -429,6 +425,24 @@ def prepare_dp_attn_batch(self, local_batch: ScheduleBatch):
...
+                        if local_batch.forward_mode.is_decode()
+                        or local_batch.forward_mode.is_idle()
+                        else 0
+                    ),
+                    dtype=torch.int32,
+                )
+                torch.distributed.all_reduce(
+                    forward_mode_state,
+                    op=torch.distributed.ReduceOp.MIN,
+                    group=self.tp_worker.get_tp_cpu_group(),
```
Comment (merrymercy) [inline python/sglang/srt/managers/scheduler.py]:
```suggestion
                    group=self.tp_cpu_group,
```

**Review by merrymercy — CHANGES_REQUESTED** [review]
